test/powershell/Language/Scripting/DeserializedTypeConversion.Tests.ps1

```diff
@@ -0,0 +1,271 @@
+
+    # Create new types and test functions.
+    $type1 = Add-Type -PassThru -TypeDefinition @'
+    public class test1
+    {
+        public string name;
+        public int port;
+        public string scriptText;
+    }
+'@    
+
+    $type2 = Add-Type -PassThru -TypeDefinition @'
+    public class test2
+    {
+        private string name;
+        private int port;
+        private string scriptText;
+
+        public string Name
+        {
+            get { return name; }
+            set { name = value; }
+        }
+
+        public int Port
+        {
+            get { return port; }
+            set { port = value; }
+        }
+
+        public string ScriptText
+        {
+            get { return scriptText; }
+            set { scriptText = value; }
+        }
+    }
+'@
+
+    $type3 = Add-Type -PassThru -TypeDefinition @'
+    public class test3
+    {
+        private string name = "default";
+        private int port = 80;
+        private string scriptText = "1..6";
+
+        public string Name
+        {
+            get { return name; }
+            set { name = value; }
+        }
+
+        public int Port
+        {
+            get { return port; }
+            set { port = value; }
+        }
+
+        public string ScriptText
+        {
+            get { return scriptText; }
+        }
+    }
+'@
+
+$type4 = Add-Type -PassThru -TypeDefinition @'
+public class test4
+{
+    private string name = "default";
+    private int port = 80;
+    private string scriptText = "1..6";
+
+    public string Name
+    {
+        get { return name; }
+        set { name = value; }
+    }
+
+    public int Port
+    {
+        get { return port; }
+        set { port = value; }
+    }
+
+    internal void Compute()
+    {
+        scriptText = scriptText + " Computed";
+    }
+}
+'@
+
+    function test-1
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test1] $test
+        )
+
+        $test | fl | Out-String
+    }    
+
+    function test-2
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test2] $test
+        )
+
+        $test | fl | Out-String
+    }
+
+    function test-3
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test3] $test
+        )
+
+        $test | fl | Out-String
+    }
+
+    function test-4
+    {
+        param(
+            [parameter(position=0, mandatory=1)]
+            [test4] $test
+        )
+
+        $test | fl | Out-String
+    }
+
+    $t1 = new-object test1 -Property @{name="TestName1";port=80;scriptText="1..5"}
+    $t2 = new-object test2 -Property @{Name="TestName2";Port=80;ScriptText="1..5"}
+    $t3 = new-object test3 -Property @{Name="TestName3";Port=80}
+    $t4 = new-object test4 -Property @{Name="TestName4";Port=80}
+    }
+    
+
+    AfterAll {        
+        Remove-Item $tempFile -force -ea silentlycontinue
+        # Clean up.
+        gsn | rsn
```

Comment:
fixed.

---
In reply to: [76680408](https://github.com/Powershell/Powershell/pull/2118#discussion_r76680408) [](ancestors = 76680408)